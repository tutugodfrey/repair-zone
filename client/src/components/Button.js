import React, {Component} from 'react';
export default class Button extends Component {
  render() {
    return  (
      <button id = {this.props.buttonId} className= {this.props.buttionClass}>
      {this.props.btnName}
      </button>
    )
  }
}