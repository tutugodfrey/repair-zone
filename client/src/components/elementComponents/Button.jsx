import React, { Component } from 'react';

export default class Button extends Component {
  render() {
    return  (
      <button
        id={this.props.buttonId}
        className={this.props.buttonClass}
        onClick={this.props.onClick}>
        {this.props.buttonName}
      </button>
    );
  }
}

class NavButton extends Button {
  render() {
    return  (
      <button
        id={this.props.buttonId}
        className={this.props.buttonClass}
        data-toggle="collapse"
        data-target="#navbar-content"
        aria-controls="narbar-content"
        aria-expanded="false"
        aria-label="Toggle-navigation"
      >
        {this.props.buttonName}
      </button>
    )
  }
}

export {
  NavButton,
}
