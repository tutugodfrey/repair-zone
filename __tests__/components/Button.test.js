import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// Component to be tested
import Button from '../../client/src/components/Button';
describe('<Button />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<Button />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});