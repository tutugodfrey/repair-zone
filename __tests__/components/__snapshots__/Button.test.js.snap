// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test <Button /> should renders without error 1`] = `
<button
  className="class value"
  onClick={[MockFunction]}
>
  name
</button>
`;

exports[`button test <NavButton /> should render without error 1`] = `
<button
  aria-controls="narbar-content"
  aria-expanded="false"
  aria-label="Toggle-navigation"
  className="class value"
  data-target="#navbar-content"
  data-toggle="collapse"
  id="unique"
>
  name
</button>
`;
